docs/topics/genericgateway.md

```diff
@@ -47,18 +64,60 @@ When calling the Generic Gateway `simpleevent` endpoint, you must include a cust
 }
 ```
 
-`Ref` and `commit` values are used to configure the specific revision that Brigade will pull from your repository, if you're using one. If both values are missing, then your Build's `ref` will be set to `master`. If you do not wish to provide any payload, just include an empty JSON object (like `{}`).
+`ref` and `commit` values are used to configure the specific revision that Brigade will pull from your repository, if you're using one. If both values are missing, your Build's `ref` will be set to `master`. Of course, you can also provide any other values you need.
+If you do not wish to provide any payload, just send an empty JSON object (`{}`). 
+
+### Calling the CloudEvent endpoint
+
+When calling the Generic Gateway `cloudevent` endpoint, you must include a valid [0.2 CloudEvent](https://github.com/cloudevents/spec/blob/v0.2/spec.md) message such as:
+
+```json
+{
+    "type":   "com.example.file.created",
+    "source": "/providers/Example.COM/storage/account#fileServices/default/{new-file}",
+    "id":     "ea35b24ede421",
+    "specversion": "0.2",
+    "data": {
+        "ref": "refs/heads/changes",
+        "commit": "63c09efb6eb544f41a48901a6d0cc6ddfa4adb28",
+        "key1": "value1",
+        "key2": "value2"
+  }
+}
+```
+
+Bear in mind that 0.2 Cloud Events specification requires non-empty and proper values for "type","source","id" and "specversion".
+
+A CloudEvent may include domain-specific information in the `data` field ([source](https://github.com/cloudevents/spec/blob/v0.2/spec.md#data-attribute)). Consequently, Brigade tries to read `ref` and `commit` values in the `data` field to configure the specific revision that Brigade will pull from your repository, if you're using one. If both values are missing, your Build's `ref` will be set to `master`.
```

Comment:
This would be great under a `Note that if you plan to use this type of event with source control ...`.